cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -176,9 +176,12 @@ function(ADD_ARROW_LIB LIB_NAME)
       BUILD_SHARED
       BUILD_STATIC
       CMAKE_PACKAGE_NAME
+      INSTALL_ARCHIVE_DIR
+      INSTALL_LIBRARY_DIR
+      INSTALL_RUNTIME_DIR
```

Comment:
> If I understand correctly these variables are new and can be used to manipulate where a library is installed?

Correct. 

> And `libarrow_python` will be installed in `arrow/python` folder now, not `arrow/python/pyarrow` if I understand correctly?

Not correct.
`libarrow_python` will be installed in `${BUILD_DIR}/pyarrow/` because we use `${BUILD_DIR}/pyarrow` for `-DCMAKE_INSTALL_PREFIX` in `setup.py` and `.` for `INSTALL_LIBRARY_DIR`:
* https://github.com/apache/arrow/pull/14925/files#diff-eb8b42d9346d0a5d371facf21a8bfa2d16fb49e213ae7c21f03863accebe0fcfR271
* https://github.com/apache/arrow/pull/14925/files#diff-005a07ca82fb46b215f58a3a8f4d6d6cc6ca088ade44e8f15cc2e6a51d1da6f4R352-R353

FYI: This is for putting artifacts to correct location (`${BUILD_DIR/pyarrow/`) by `cmake --build --target install` instead of copying in `setup.py`.

Reply:
Thank you for the explanation :pray: